core/rendered_connection.ts

```diff
@@ -566,3 +547,25 @@ export class RenderedConnection extends Connection {
     }
   }
 }
+
+export namespace RenderedConnection {
+  /**
+   * Enum for different kinds of tracked states.
+   *
+   * WILL_TRACK means that this connection will add itself to
+   * the db on the next moveTo call it receives.
+   *
+   * UNTRACKED means that this connection will not add
+   * itself to the database until setTracking(true) is explicitly called.
+   *
+   * TRACKED means that this connection is currently being tracked.
+   */
+  export enum TrackedState {
+    WILL_TRACK = -1,
+    UNTRACKED = 0,
+    TRACKED = 1,
+  }
+}
+
+export type TrackedState = RenderedConnection.TrackedState;
+export const TrackedState = RenderedConnection.TrackedState;
```

Comment:
dunno how you're handling formatting the new files, but at some point i assume a lint rule somewhere will complain about this lack of new line at end of file

Reply:
Right now clang-format is properly running, but it looks like the eslint config was never updated to parse the typescript. Will fix the problems in this PR, and then investigate separately.